src/Plugin/Visualisation/Style/AxisChart.php

```diff
@@ -836,4 +857,37 @@ protected function setArrayOrder(array $array_to_order, $weight_key = 'weight')
     return $array_to_order;
   }
 
+  /**
+   * Returns the URI of a DVF file or dataset (E.g. for downloading).
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The Drupal entity that contains the dvf_url or dvf_file.
+   *
+   * @return string
+   *   The URI of the file or dataset.
+   */
+  protected function getDatasetDownloadUri($entity) {
```

Comment:
@rob-duplock  Does this account for future possible uris where dataset might be a JSON endpiont, view, or dataset that does not have a download link?

Maybe an additional method should be added to identify if something is downloadable and that determines button visibility and is also guardcode for a method like this one